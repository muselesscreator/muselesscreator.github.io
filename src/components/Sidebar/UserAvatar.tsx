import contactInfo from '~/data/contact';

const UserAvatar = () => (
  <figure className="avatar-box">
    <img src={contactInfo.avatar} alt={contactInfo.name} width="80" />
  </figure>
);

export default UserAvatar;
