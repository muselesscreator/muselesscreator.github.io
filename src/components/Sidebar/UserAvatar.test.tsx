import {
  describe,
  test,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';
import contactInfo from '~/data/contact';

import UserAvatar from './UserAvatar';

describe('UserAvatar component', () => {
  test('snapshot includes user avatar', () => {
    const el = shallow(<UserAvatar />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance.findByType('img')[0].props.src).toBe(contactInfo.avatar);
  });
});
