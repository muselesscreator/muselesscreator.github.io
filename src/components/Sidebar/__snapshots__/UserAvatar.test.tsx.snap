// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserAvatar component > snapshot includes user avatar 1`] = `
"<figure
  className="avatar-box"
>
  <img
    alt="Ben Warzeski"
    src="/src/assets/ampersand.png"
    width="80"
  />
</figure>"
`;
